src/coffea/dataset_tools/apply_processor.py

```diff
@@ -0,0 +1,38 @@
+import copy
+from typing import Callable, Union
+
+from coffea.nanoevents import NanoAODSchema, NanoEventsFactory
+from coffea.processor import ProcessorABC
+
+
+def apply_to_one_dataset(
+    data_manipulation: Union[ProcessorABC, Callable],
```

Comment:
`function`, maybe in docs it can discuss the function needs to be traceable via dask_awkward

Reply:
```python3
def f():
    pass
print(isinstance(f, Callable))
```
prints `True` and other folks may want to implement functors with state and such, so Callable seemed most appropriate.

We could even restrict to 
```python3
Callable[
    [dask_awkward.Array], 
    Tuple[Union[
        dask.base.DaskMethodsMixin,
        Dict[str, dask.base.DaskMethodsMixin],
        Set[dask.base.DaskMethodsMixin],
        List[dask.base.DaskMethodsMixin],
        Tuple[dask.base.DaskMethodsMixin],
    ], ...]
]
```

Reply:
From discussion in person I'll leave a comment in the code, it should be sufficient.